File: lab2_task1/main.kt
fun checkBoardBounds(n: Int, m: Int, steps: String): String {
    var startX = 1
    while (startX <= n) {
        var startY = 1
        while (startY <= m) {
            var x = startX
            var y = startY
            var isValid = true
            var i = 0
            while (i < steps.length && isValid) {
                val step = steps[i]
                when (step) {
                    'L' -> x -= 1
                    'R' -> x += 1
                    'D' -> y -= 1
                    'U' -> y += 1
                    else -> return "Ошибка в ходах"
                }
                if (x < 1 || x > n || y < 1 || y > m) {
                    isValid = false
                }
                i += 1
            }
            if (isValid) {
                return "($startX,$startY)"
            }
            startY += 1
        }
        startX += 1
    }
    return "No"
}

fun main() {
    val n = readLine()!!.toInt()
    val m = readLine()!!.toInt()
    val steps = readLine()!!.trim()
    println(checkBoardBounds(n, m, steps))
}

File: lab2_task1/main.rs
use std::io;

fn check_board_bounds(n: i32, m: i32, steps: &str) -> String {
    for start_x in 1..=n {
        for start_y in 1..=m {
            let (mut x, mut y) = (start_x, start_y);
            let mut is_valid = true;
            for step in steps.chars() {
                match step {
                    'L' => x -= 1,
                    'R' => x += 1,
                    'D' => y -= 1,
                    'U' => y += 1,
                    _ => return "Ошибка в ходах".to_string(),
                }
                if x < 1 || x > n || y < 1 || y > m {
                    is_valid = false;
                    break;
                }
            }
            if is_valid {
                return format!("({},{})", start_x, start_y);
            }
        }
    }
    "No".to_string()
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let n: i32 = input.trim().parse().unwrap();
    
    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let m: i32 = input.trim().parse().unwrap();
    
    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let steps = input.trim();
    
    println!("{}", check_board_bounds(n, m, steps));
}